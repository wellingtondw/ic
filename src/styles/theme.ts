export default {
  border: {
    radius: '0.625rem'
  },
  font: {
    main: {
      family: "'Montserrat', sans-serif",
      normal: 400,
      bold: 700
    },
    secondary: {
      family: "'Lato', sans-serif",
      bold: 700
    },
    sizes: {
      xsmall: '0.75rem',
      small: '0.875rem',
      normal: '1rem',
      medium: '1.125rem',
      large: '2.25rem',
      xlarge: '4.5rem'
    }
  },
  colors: {
    primary: '#2C97D1',
    black: '#2A2A2A',
    blackBg: '#2B2B2B',
    white: '#FFFFFF',
    yellow: '#FBFE63'
  },
  spacings: {
    xxxsmall: '0.25rem', // 4px
    xxsmall: '0.5rem', //8px
    xsmall: '1rem', //16px
    small: '1.5rem', //24px
    medium: '2rem', // 32px
    large: '2.5rem', //40px
    xlarge: '3rem', //48px
    xxlarge: '3.5rem' //56px
  },
  transition: {
    default: '0.3s ease-in-out',
    fast: '0.1s ease-in-out'
  }
} as const
