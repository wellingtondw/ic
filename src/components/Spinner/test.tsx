import { renderWithTheme } from '../../utils/tests/helpers'
import theme from '../../styles/theme'
import { darken } from 'polished'

import { Spinner } from '.'

describe('<Spinner />', () => {
  it('should be able to render correctly', () => {
    const { container } = renderWithTheme(<Spinner />)

    expect(container.firstChild).toMatchSnapshot()
  })

  it('should be able to render with primary variant', () => {
    const { container } = renderWithTheme(<Spinner />)

    expect(container.firstChild).toHaveStyle({
      'border-top': `1px solid ${theme.colors.white}`,
      'border-right': `1px solid ${theme.colors.white}`,
      'border-bottom': `1px solid ${theme.colors.white}`,
      'border-left': `2px solid ${theme.colors.primary}`
    })
  })

  it('should be able to render with secondary variant', () => {
    const { container } = renderWithTheme(<Spinner variant="secondary" />)

    expect(container.firstChild).toHaveStyle({
      'border-top': `1px solid ${theme.colors.primary}`,
      'border-right': `1px solid ${theme.colors.primary}`,
      'border-bottom': `1px solid ${theme.colors.primary}`,
      'border-left': `2px solid ${darken(0.2, theme.colors.primary)}`
    })
  })
})
