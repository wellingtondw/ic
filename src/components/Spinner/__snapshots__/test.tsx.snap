// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Spinner /> should be able to render correctly 1`] = `
.c0 {
  -webkit-animation: fvtopB 1s linear infinite;
  animation: fvtopB 1s linear infinite;
  -webkit-transform: translateZ(0);
  -ms-transform: translateZ(0);
  transform: translateZ(0);
  background: transparent;
  width: 1.125rem;
  height: 1.125rem;
  border-radius: 50%;
  border-top: 1px solid #FFFFFF;
  border-right: 1px solid #FFFFFF;
  border-bottom: 1px solid #FFFFFF;
  border-left: 2px solid #2C97D1;
}

<div
  class="c0"
/>
`;
